aten/src/ATen/native/Linear.cpp

```diff
@@ -548,8 +548,9 @@ Tensor _trilinear(const Tensor& i1_, const Tensor& i2_, const Tensor& i3_,
   auto output = at::zeros(output_size, i1.options());
 
   // Three conditionals are necessary since this function is meant to work for both
-  // forward and backward, which changes the dimensions of the inputs.
-  if (output.numel() != 0 && i1.numel() != 0 && i2.numel() != 0) {
+  // forward and backward, which changes the dimensions of the inputs. The output
+  // will have 0 dim even if a single input has zero dim.
```

Comment:
Ugh, it's so difficult to write these things. The opposite direction of the implication is the one that is true.
```suggestion
  // forward and backward, which changes the dimensions of the inputs.
  // Note that if output.numel() had zero elems is because (at least) one of i1, i2, i3 had zero elems.
```